### PR TITLE
set timezone for query parser, add tests

### PR DESCRIPTION
**Pull request by LenoreChen**
This PR is to address timezone issue when parsing queries regarding datetime
* set timezone to UTC in `QueryParser`, similar to [`FScriptParser`](https://github.com/kgrgreer/foam3/blob/development/src/foam/parse/FScriptParser.java#L484)
* add tests for parsing and evaluating datetime

nanopay ci: https://ci-jenkins.nanopay.net/view/CI/job/ci-NP-8303-repoDocDateFormat/1/console

More detailed description:
when parsing queries, if timezone is not set to UTC, the dates parsed are set to local timezone by default, which causes errors when comparing with properties

### CODE REVIEW

File: src/foam/parse/QueryParserUserTest.js [hunk 1]
```diff
@@ -74,31 +84,64 @@ foam.CLASS({
       test( isValid("firstName=Simon,Wassim"," ( firstname =  ?  )  OR  ( firstname =  ?  ) ") , "The firstName is equal to value1 OR to value2");
       //          {"id=(6|7)"," ( ( ( id =  ?  )  OR  ( id =  ?  ) ) ) "},
       //          {"id=(6|7)"," ( ( ( id =  ?  )  OR  ( id =  ?  ) ) ) "},//TODO add alises
+
+      // Parse query and evalue predicates on user
```
Comment (kgrgreer) [inline src/foam/parse/QueryParserUserTest.js]: evalue -> evaluate ?

Comment (LenoreChen) [reply src/foam/parse/QueryParserUserTest.js]: corrected

File: src/foam/parse/QueryParser.java [hunk 2]
```diff
@@ -488,7 +488,9 @@ private Grammar getGrammar() {
         dateFmt >=  5 ? dateFmt == 5 ? (Integer) result[4] + 1 : (Integer) result[4]  : 0,
         dateFmt >=  7 ? dateFmt == 7 ? (Integer) result[6] + 1 : (Integer) result[6]  : 0,
         dateFmt >=  9 ? dateFmt == 9 ? (Integer) result[8] + 1 : (Integer) result[8]  : 0);
-
+      
+      start.setTimeZone(TimeZone.getTimeZone("UTC"));
```
Comment (jlhughes) [inline src/foam/parse/QueryParser.java:493]: do we want to set the timezone or convert to the timezone?

**Comment by jlhughes** [comment]
Also, QueryParserUserTest moved to parse/test

**Review by jlhughes — APPROVED** [review]

**Pull request merged**